pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1045,19 +1047,26 @@ public CompletableFuture<Optional<Topic>> getTopic(final TopicName topicName, bo
             }
             final boolean isPersistentTopic = topicName.getDomain().equals(TopicDomain.persistent);
             if (isPersistentTopic) {
-                return topics.computeIfAbsent(topicName.toString(), (tpName) -> {
-                    if (topicName.isPartitioned()) {
-                        return fetchPartitionedTopicMetadataAsync(TopicName.get(topicName.getPartitionedTopicName()))
-                                .thenCompose((metadata) -> {
-                                    // Allow crate non-partitioned persistent topic that name includes `partition`
-                                    if (metadata.partitions == 0
-                                            || topicName.getPartitionIndex() < metadata.partitions) {
-                                        return loadOrCreatePersistentTopic(tpName, createIfMissing, properties);
-                                    }
-                                    return CompletableFuture.completedFuture(Optional.empty());
-                                });
-                    }
-                    return loadOrCreatePersistentTopic(tpName, createIfMissing, properties);
+                final CompletableFuture<Optional<TopicPolicies>> topicPoliciesFuture =
+                        getTopicPoliciesBypassSystemTopic(topicName);
+                return topicPoliciesFuture.thenCompose(optionalTopicPolicies -> {
+                    final TopicPolicies topicPolicies = optionalTopicPolicies.orElse(null);
```

Comment:
Why not pass the `optionalTopicPolicies` directly. So that you don't need to add @Nullable to the method signature https://github.com/apache/pulsar/pull/21445/files#diff-0210356c8a88e4efa89eb769a027fa6c166db479dbad8bbbbc704c6ed6e317f5R1523

Reply:
Passing the `Optional` is not a good approach. It will force the caller to create an extra Object. Plus, refer to the JDK doc of Optional. https://github.com/mattisonchao/jdk/blob/f44032969f37948a49e5bd0ed6738aa4d93687da/src/java.base/share/classes/java/util/Optional.java#L52